src/cmd/flux-account-service.py

```diff
@@ -0,0 +1,422 @@
+from fluxacct.accounting import queue_subcommands as qu
+from fluxacct.accounting import project_subcommands as p
+
+
+def establish_sqlite_connection(path):
+    # try to open database file; will exit with -1 if database file not found
+    if not os.path.isfile(path):
+        print(f"Database file does not exist: {path}", file=sys.stderr)
+        sys.exit(1)
+
+    db_uri = "file:" + path + "?mode=rw"
+    try:
+        conn = sqlite3.connect(db_uri, uri=True)
+        # set foreign keys constraint
+        conn.execute("PRAGMA foreign_keys = 1")
+    except sqlite3.OperationalError as exc:
+        print(f"Unable to open database file: {db_uri}", file=sys.stderr)
+        print(f"Exception: {exc}")
+        sys.exit(1)
+
+    return conn
+
+
+def background():
+    pid = os.fork()
+    if pid > 0:
+        # exit first parent
+        sys.exit(0)
+
+
+def check_db_version(conn):
+    # check version of database; if not up to date, output message
+    # and exit
+    cur = conn.cursor()
+    cur.execute("PRAGMA user_version")
+    db_version = cur.fetchone()[0]
+    if db_version < 20:
+        print(
+            "flux-accounting database out of date; please update DB with 'flux account-update-db' before running commands"
+        )
+        sys.exit(1)
+
+
+# pylint: disable=broad-except
+class AccountingService:
+    def __init__(self, flux_handle, conn):
+
+        self.handle = flux_handle
+        self.conn = conn
+
+        try:
+            # register service with broker
+            self.handle.service_register("accounting").get()
+            print("registered accounting service", file=sys.stderr)
+        except FileExistsError as exc:
+            logging.error("flux-accounting service is already registered")
+
+        # register signal watcher for SIGTERM to initiate shutdown
+        self.handle.signal_watcher_create(signal.SIGTERM, self.shutdown).start()
+        self.handle.signal_watcher_create(signal.SIGINT, self.shutdown).start()
+
+        endpoints = [
+            "view_user",
+            "add_user",
+            "delete_user",
+            "edit_user",
+            "view_bank",
+            "add_bank",
+            "delete_bank",
+            "edit_bank",
+            "view_job_records",
+            "update_usage",
+            "add_queue",
+            "view_queue",
+            "delete_queue",
+            "edit_queue",
+            "add_project",
+            "view_project",
+            "delete_project",
+            "shutdown",
+        ]
+
+        for name in endpoints:
+            self.handle.msg_watcher_create(
+                getattr(self, name), FLUX_MSGTYPE_REQUEST, f"accounting.{name}", self
+            ).start()
+
+    def shutdown(self, handle, watcher, signum, arg):
+        print("Shutting down...", file=sys.stderr)
+        self.conn.close()
+        self.handle.service_unregister("accounting").get()
+        self.handle.reactor_stop()
+
+    def view_user(self, handle, watcher, msg, arg):
```

Comment:
Yeah, the behavior of the `view-user` command doesn't fail if the user can't be found in the flux-accounting DB, it just returns `User not found in association_table` if it can't find it. This has always been the behavior, but I can definitely open an issue on this and work on this on the side. 👍 I'll change it to instead return a non-zero exit code and perhaps log an error message that it could not find the user.